xarray/backends/netCDF4_.py

```diff
@@ -640,6 +646,7 @@ def open_dataset(  # type: ignore[override]  # allow LSP violation, not supporti
         persist=False,
         lock=None,
         autoclose=False,
+        memory: bytes | None = None,  # type: ignore[override]
```

Comment:
We really should fix these liskov issues at some point.....
Not your fault though!